base/libgit2/consts.jl

```diff
@@ -271,12 +274,14 @@ Option flags for `GitRepo`.
                         CREDTYPE_USERNAME           = Cuint(1 << 5),
                         CREDTYPE_SSH_MEMORY         = Cuint(1 << 6))
 
+if LibGit2.version() >= v"0.24.0"
```

Comment:
Currently, `libgit2` is still a bit of flux. I would recommend minimum version 0.23. Question, why do we update to 0.24? Is that because of `git_stream_register_tls`?